src/mesh/voxel_cloud.rs

```diff
@@ -463,6 +481,322 @@ impl ScalarField {
+        if bounding_box_other == None {
+            return;
+        }
+
+        let bounding_boxes = [bounding_box_self, bounding_box_other];
+
+        // Unwrap the bounding box options. the other bounding box must be valid
+        // at this point and the self can be None. In that case, all the volume
+        // voxels from the other scalar field will be remapped to the current
+        // scalar field.
+        let valid_bounding_boxes_iter = bounding_boxes.iter().filter_map(|b| *b);
+
+        if let Some(bounding_box) = BoundingBox::union(valid_bounding_boxes_iter) {
+            // Resize (keep or grow) the current scalar field to a block that
+            // will contain union voxels.
+            self.resize_to_voxel_space_bounding_box(&bounding_box);
+
+            // Iterate through the block of space containing volume voxels from
+            // both scalar fields. Iterate through the units of the current
+            // scalar field.
+            for (one_dimensional, voxel) in self.voxels.iter_mut().enumerate() {
+                // If the current scalar field doesn't contain a volume voxel at
+                // the current position
+                if !is_voxel_within_range(*voxel, volume_value_range_self) {
+                    let cartesian_coordinate = one_dimensional_to_cartesian_coordinate(
+                        one_dimensional,
+                        &self.block_start,
+                        &self.block_dimensions,
+                        &self.voxel_dimensions,
+                    );
+                    let absolute_coordinate_other = cartesian_to_absolute_voxel_coordinate(
+                        &cartesian_coordinate,
+                        &other.voxel_dimensions,
+                    );
+
+                    // If the other scalar field contains a voxel on the
+                    // cartesian coordinate of the current voxel, then remap the
+                    // other value to the volume value range of the current
+                    // scalar field and set the voxel to the value.
+                    if let Some(voxel_other) =
+                        other.value_at_absolute_voxel_coordinate(&absolute_coordinate_other)
+                    {
+                        if volume_value_range_other.contains(&voxel_other) {
+                            // If the remap fails, the program should panic.
+                            *voxel = Some(
+                                math::remap(
+                                    voxel_other,
+                                    volume_value_range_other,
+                                    volume_value_range_self,
+                                )
+                                .expect("One of the ranges is infinite.")
+                                    as f32,
+                            );
+                        }
+                    }
+                }
+            }
+            self.shrink_to_fit(volume_value_range_self);
```

Comment:
true. the same applies for the intersection

Reply:
Hmmm, I am not sure it applies to intersection. The intersection bounding box may and up bigger than the actual volume, because voxels might be turned afterwards.